### PR TITLE
Add reviewing aggregate, commands and events

### PR DESCRIPTION
**Pull request by timpeat**
## Description of change

Adds Review aggregates and reviewing commands and events.

adds the Reviewing::Review aggregate root (event sourced).

adds the following commands:
Reviewing::ReceiveApplication
Reviewing::SendBack
Reviewing::Complete

adds the following events:
Reviewing::ApplicationReceived
Reviewing::SentBack
Reviewing::Completed

adds a new type, Types::ReturnReason

## Link to relevant ticket

[CRIMRE-84](https://dsdmoj.atlassian.net/browse/CRIMRE-84)

## Notes for reviewer

In order to build the time passed reports (and many other things), Review needs a temporal understanding of changes in an applications review status.

## How to manually test the feature

[CRIMRE-84]: https://dsdmoj.atlassian.net/browse/CRIMRE-84?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -69,3 +69,7 @@ Rails/SkipsModelValidations:
   Exclude:
     - '**/app/read_models/**/*'
     - '**/lib/warden/**/*'
+
```
Comment (timpeat) [inline .rubocop.yml:72]: Rubocop doesn't care which one you use, only that you are consistent. Here I've changed to block as it is more versatile when the object needs reloading or methods chaining.

File: app/aggregates/reviewing.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module Reviewing
+  %w[event command].each do |sub_dir|
+    require File.expand_path("reviewing/#{sub_dir}.rb", __dir__)
```
Comment (willmcb) [inline app/aggregates/reviewing.rb:3]: Is it worth breaking rails conventions? Zweitwork etc.

File: app/aggregates/reviewing/commands/send_back.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module Reviewing
+  class SendBack < Command
+    attribute :application_id, Types::Uuid
+    attribute :user_id, Types::Uuid
+    attribute :reason, Types::ReturnReason
```
Comment (willmcb) [inline app/aggregates/reviewing/commands/send_back.rb:5]: Eventually reasons probably need to be on the datastore

**Review by willmcb — APPROVED** [review]
👍

**Pull request merged**